core/src/km_core_processevent_api.cpp

```diff
@@ -52,6 +52,27 @@ km_core_process_event(km_core_state *state,
   }
   km_core_status status = state->processor().process_event(state, vk, modifier_state, is_key_down, event_flags);
 
+  if (state_should_invalidate_context(state, vk, modifier_state, is_key_down, event_flags)) {
+    state->context().clear();
+    // we are already committed. So we need to un-commit (remove the end of the vector)
+    if (state->actions().back().type == KM_CORE_IT_END) {
+      state->actions().pop_back();
+    }
+    bool foundEmit = false;
+    km::core::action oldAction;
+    // try to put the invalidate item before the emit keystroke
+    if (state->actions().back().type == KM_CORE_IT_EMIT_KEYSTROKE) {
+      oldAction = state->actions().back();
+      state->actions().pop_back();
+      foundEmit = true;
+    }
+    state->actions().push_invalidate_context();
+    if (foundEmit) {
+      state->actions().push_back(oldAction);
+    }
+    state->actions().commit();
+  }
+
   state->apply_actions_and_merge_app_context();
```

Comment:
I thought since we had already had the run through the processors rules (LDML, KMX) this was the correct place to able to clear the context. However for the LDML normalize case we where not quite done. For the ldml case `apply_actions_and_merge_app_context()` will result in a call `action_item_list_to_actions_object()` and this is now the context been cleared it results in update the `actions.code_points_to_delete` to delete all application context.

Reply:
@srl295  we may want to put this PR back in Draft 
@mcdurdin  It is not ready for re-review I was trying to remove my approval but instead requested a re-review.

Reply:
@rc-swag i'll reply on the body of the PR